src/layout/ModalField.component.js

```diff
@@ -0,0 +1,245 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    email as emailValidator,
+    InputField,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+    NoticeBox,
+    Tooltip,
+} from '@dhis2/ui'
+import TextField from 'd2-ui/lib/form-fields/TextField'
+import PropTypes from 'prop-types'
+import React, { useMemo, useState } from 'react'
+import styles from './ModalField.component.module.css'
+import { VerifyEmail } from './VerifyEmail.component.js'
+
+const TooltipWrapper = ({ disabled, content, children }) => {
+    if (!disabled) {
+        return <>{children}</>
+    }
+    return <Tooltip content={content}>{children}</Tooltip>
+}
+
+TooltipWrapper.propTypes = {
+    children: PropTypes.node,
+    content: PropTypes.string,
+    disabled: PropTypes.bool,
+}
+
+const getSaveDisabledContent = ({ newEmail, emailValidationMessage }) => {
+    if (!newEmail) {
+        return i18n.t('No email provided')
+    }
+    if (emailValidationMessage) {
+        return i18n.t('Email is invalid')
+    }
+    return i18n.t('Emails must match')
+}
+
+const RemoveModal = ({
+    removeModalOpen,
+    closeModal,
+    userEmailVerified,
+    setUserEmail,
+    onUpdate,
+}) => (
+    <Modal hide={!removeModalOpen} onClose={closeModal}>
+        <ModalTitle>{i18n.t('Remove email')}</ModalTitle>
+
+        <ModalContent>
+            {userEmailVerified && (
+                <NoticeBox
+                    className={styles.emailModalItem}
+                    title={i18n.t('Your email is currently verified')}
+                    warning
+                ></NoticeBox>
+            )}
+            <div>{i18n.t('Are you sure you want to remove your email?')}</div>
+        </ModalContent>
+
+        <ModalActions>
+            <ButtonStrip end>
+                <Button onClick={() => closeModal()} secondary>
+                    {i18n.t('Cancel')}
+                </Button>
+
+                <Button
+                    onClick={() => {
+                        setUserEmail(null)
+                        onUpdate('email', '')
+                        closeModal()
+                    }}
+                    destructive
+                >
+                    {i18n.t('Remove email')}
+                </Button>
+            </ButtonStrip>
+        </ModalActions>
+    </Modal>
+)
+
+RemoveModal.propTypes = {
+    closeModal: PropTypes.func,
+    removeModalOpen: PropTypes.bool,
+    setUserEmail: PropTypes.bool,
+    userEmailVerified: PropTypes.bool,
+    onUpdate: PropTypes.func,
+}
+
+export function ModalField({
```

Comment:
I'm a bit confused about the naming of this. When I read "ModalField" I would expect this to be a generic field that somehow opens a modal. I would expect this to be called `EmailField`.

Reply:
I've updated to use EmailField, and to make some of the related functions/props correspond to this update